src/pivot-table/components/cells/ColumnAdjuster.tsx

```diff
@@ -14,47 +13,41 @@ interface AdjusterProps {
   dataModel: DataModel | undefined;
   isLastColumn: boolean;
 }
-
 /**
  * Component that is placed on top of column border, to resize the columns.
  * When you start dragging, mouse move and mouse up listeners are added.
  * While dragging this components follows the pointer, and on mouse up all column widths are updated.
  */
 const ColumnAdjuster = ({ cellInfo, columnWidth, dataModel, isLastColumn }: AdjusterProps) => {
-  const [internalWidth, setInternalWidth] = useState(columnWidth);
-  const tempWidth = useRef({ initWidth: 0, columnWidth: 0, initX: 0 });
+  const [, setInternalWidth] = useState(columnWidth);
   const positionAdjustment = isLastColumn ? CELL_PADDING : CELL_PADDING + GRID_BORDER;
 
-  useOnPropsChange(() => {
-    setInternalWidth(columnWidth);
-  }, [columnWidth]);
+  const tempWidth = useMemo(() => ({ initWidth: 0, columnWidth, initX: 0 }), [columnWidth]);
```

Comment:
Change this to `const tempWidth = useMemo(() => ({ initWidth: columnWidth, columnWidth, initX: 0 }), [columnWidth]);` and you dont need to assign values in the `mouseDownHandler`.
```js
// These can be removed
tempWidth.initWidth = columnWidth;
 tempWidth.columnWidth = columnWidth;
```